### DISCUSSION TITLE
Using DOLFYN Module for Nortek Aquadopp Data

### DISCUSSION THREAD

**Discussion by MogoBRE** [MHKiT-Software/MHKiT-Python#356] [Q&A] [answered]
Hello. I am curious if the DOLFYN module has the capability to read Nortek binary data coming from an Aquadopp ADCP, which comes in as a **.prf**. I am getting a 'config_type' KeyError, which seems to indicate that this file format is not supported, but wanted to ensure that it is not due to some other mistake I might be making. Thank you.

**Comment by ssolson** [comment]
@jmcvey3 do you have any thoughts on this discussion?

**Reply by jmcvey3** [reply, marked as answer]
@MogoBRE Apologies I never saw this, we haven't yet integrated the aquadopp because we typically don't use them. I see the info in the Nortek integrator guide so it shouldn't be hard to add. Please toss this up as an issue, and if you can provide a small (<=1 mb) clip of a datafile that would be helpful.

**Comment by akeeste** [comment]
@MogoBRE if this is still of interest and you can share some sample data, we could move forward with adding this functionality

**Comment by MRCGit-collab** [comment]
Hi, this would be really valuable. I can include a sample file here. AQD was collecting in burst mode
[Git_example.zip](https://github.com/user-attachments/files/24916118/Git_example.zip)

**Reply by MogoBRE** [reply]
Hi. I apologize for not following up on this. I muscled my way through the data analysis I had to do and then had to move on to other projects. 
Attached is 1 of 2 folders with data from a deployment we did in Humboldt Bay, CA. This is the smallest data set I have (~32MB compressed). Hope this is helpful and happy to answer any questions you might have.
[BladeRunner Energy_Humboldt Bay.zip](https://github.com/user-attachments/files/24919112/BladeRunner.Energy_Humboldt.Bay.zip)

**Reply by MogoBRE** [reply]
2nd folder....
[BladeRunner Energy_Humboldt Bay_2nd.zip](https://github.com/user-attachments/files/24919127/BladeRunner.Energy_Humboldt.Bay_2nd.zip)

**Reply by MRCGit-collab** [reply]
Thanks for taking this up!

**Reply by jmcvey3** [reply]
Thank you both for the files, I'll see how hard this will be to implement.

**Reply by jmcvey3** [reply]
@MRCGit-collab @MogoBRE Please see PR #434, pull it, and try running your .prf files. 
@MRCGit-collab The cell size and blanking distance are specified differently for the high resolution profiling, so the range coordinate will not be accurate in the datafile you get. Until I figure out how to fix that, you can see the .hdr file for the correct values to manually recreate the range coordinate.

**Reply by jmcvey3** [reply]
Replying to note that I manually back-calculated a set of coefficients for determining the high-resolution profile range-bins from the files available, and this seems to work well. All of the changes from this discussion were included in the v1.1.0 release
